src/module.api/loan.vault.service.ts

```diff
@@ -139,20 +144,28 @@ export class LoanVaultService {
       batchCount: data.batchCount,
       liquidationHeight: data.liquidationHeight,
       liquidationPenalty: data.liquidationPenalty,
-      batches: await this.mapLiquidationBatches(data.batches)
+      batches: await this.mapLiquidationBatches(data)
     }
   }
 
-  private async mapLiquidationBatches (batches: VaultLiquidationBatch[]): Promise<LoanVaultLiquidationBatch[]> {
-    if (batches.length === 0) {
+  private async mapLiquidationBatches (data: VaultLiquidation): Promise<LoanVaultLiquidationBatch[]> {
+    if (data.batches.length === 0) {
       return []
     }
 
-    const items = batches.map(async batch => {
+    const end = data.liquidationHeight
+    const start = end - this.NUM_AUCTION_EXP_BLOCKS
+
+    const items = data.batches.map(async batch => {
+      const lt = `${HexEncoder.encodeHeight(end)}-${'f'.repeat(64)}`
+      const gt = `${HexEncoder.encodeHeight(start)}-${'0'.repeat(64)}`
+      const bids = await this.vaultAuctionHistoryMapper.query(`${data.vaultId}-${batch.index}`, Number.MAX_SAFE_INTEGER, lt, gt)
+
       return {
         index: batch.index,
         collaterals: await this.mapTokenAmounts(batch.collaterals),
         loan: (await this.mapTokenAmounts([batch.loan]))[0],
+        froms: bids.length > 0 ? bids.map(b => b.from) : [],
```

Comment:
```suggestion
        froms: bids.map(b => b.from),
```

Since you are checking length, I assume you don't need this?

Reply:
I think we should also map this into human-readable address instead of leaving it as script.